core-bundle/src/Resources/contao/modules/Module.php

```diff
@@ -499,9 +444,13 @@ protected function compileNavigationRow(PageModel $objPage, PageModel $objSubpag
 	 * @param boolean $blnIsSitemap  If true, the sitemap settings apply
 	 *
 	 * @return array<array{page:PageModel,hasSubpages:bool}>|null
+	 *
+	 * @deprecated Will be removed in Contao 5.0, no replacement is given. This method was moved to FrontendMenuBuilder#findPagesByPid().
```

Comment:
Then `FrontendMenuBuilder::findPagesByPid` is the replacement, is it not?

Reply:
I don't know why we would want to make this method public and static.

Reply:
Also see contao/contao#2935.

Reply:
It does not need to be static. If you are referring to my usage of `::` - I am not sure using `#` to denote object methods is common? Even PHP uses `::` for everything.

Reply:
If we want others to use this method, we would need to extract this method into some kind of repository. Here, it should stay an internal method IMHO.

Reply:
@fritzmg Here is an example from the Doctrine docs where `#` is used to denote object context: https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/query-builder.html#sql-query-builder I kind of like this :slightly_smiling_face:

Reply:
But `__METHOD__` always uses `::` and you use `parent::` in object methods as well ;)

Reply:
Yes, and you can (still) use `->` to call static methods. These things … ehm … "haven't developed yet". :smile:

Reply:
> If we want others to use this method, we would need to extract this method into some kind of repository. Here, it should stay an internal method IMHO.

Yes, just saysing I am not sure whether we should really deprecate a public method without providing an alternative.